pkg/templatelib/lib.go

```diff
@@ -68,55 +69,33 @@ func stringsModifierActionFactory(a func(string, string) string) func([]string,
 	}
 }
 
-var FuncMap = template.FuncMap{
-	// {{- $isGitHub := hasPrefix "https://github.com/" $url -}}
-	// {{- $isHtml := hasSuffix ".html" $url -}}
-	"hasPrefix": swapStringsFuncBoolArgsOrder(strings.HasPrefix),
-	"hasSuffix": swapStringsFuncBoolArgsOrder(strings.HasSuffix),
-
-	// {{- $hugeIfTrue := .SomeValue | ternary "HUGE" "not so huge" -}}
-	// if .SomeValue is truthy, $hugeIfTrue will be "HUGE"
-	// (otherwise, "not so huge")
-	"ternary": func(truthy interface{}, falsey interface{}, val interface{}) interface{} {
-		if t, ok := template.IsTrue(val); !ok {
-			panic(fmt.Sprintf(`template.IsTrue(%+v) says things are NOT OK`, val))
-		} else if t {
-			return truthy
-		} else {
-			return falsey
+func FuncMap() template.FuncMap {
+	funcMap := sprig.TxtFuncMap()
+
+	// https://github.com/Masterminds/sprig/pull/276
+	funcMap["ternary"] = func(vt interface{}, vf interface{}, v interface{}) interface{} {
+		if truth, ok := template.IsTrue(v); !ok {
+			panic(fmt.Sprintf(`template.IsTrue(%+v) says things are NOT OK`, v))
+		} else if truth {
+			return vt
 		}
-	},
+		return vf
+	}
 
-	// First Tag: {{- .Tags | first -}}
-	// Last Tag:  {{- .Tags | last -}}
-	"first": thingsActionFactory("first", true, func(args []interface{}, arg interface{}) interface{} { return arg }),
-	"last":  thingsActionFactory("last", false, func(args []interface{}, arg interface{}) interface{} { return arg }),
+	// Everybody: {{- join ", " .Names -}}
+	// Concat: {{- join "/" "https://github.com" "jsmith" "some-repo" -}}
+	funcMap["join"] = stringsActionFactory("join", true, strings.Join)
+	// (this differs slightly from the Sprig "join" in that it accepts either a list of strings or multiple arguments - Sprig instead has an explicit "list" function which can create a list of strings *from* a list of arguments so that multiple-signature usability like this is not necessary)
```

Comment:
I guess in looking at https://masterminds.github.io/sprig/string_slice.html, the key here will be to verify that we can start porting to the `list ... | join ...` syntax with this implementation so that at some point we can drop this.